WebService/v1/Controllers/SimulationsController.cs

```diff
@@ -18,34 +21,69 @@ namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controller
     public class SimulationsController : Controller
     {
         private const int MAX_DELETE_DEVICES = 100;
+
         private readonly ISimulations simulationsService;
+        private readonly IServicesConfig servicesConfig;
+        private readonly IDeploymentConfig deploymentConfig;
         private readonly IIotHubConnectionStringManager connectionStringManager;
+        private readonly IPreprovisionedIotHub preprovisionedIotHub;
         private readonly ISimulationAgent simulationAgent;
-
+        private readonly ISimulationRunner simulationRunner;
+        private readonly IRateLimiting rateReporter;
         private readonly ILogger log;
 
         public SimulationsController(
             ISimulations simulationsService,
+            IServicesConfig servicesConfig,
+            IDeploymentConfig deploymentConfig,
             IIotHubConnectionStringManager connectionStringManager,
+            IPreprovisionedIotHub preprovisionedIotHub,
             ISimulationAgent simulationAgent,
+            ISimulationRunner simulationRunner,
+            IRateLimiting rateReporter,
             ILogger logger)
         {
             this.simulationsService = simulationsService;
+            this.servicesConfig = servicesConfig;
+            this.deploymentConfig = deploymentConfig;
             this.connectionStringManager = connectionStringManager;
+            this.preprovisionedIotHub = preprovisionedIotHub;
             this.simulationAgent = simulationAgent;
+            this.simulationRunner = simulationRunner;
+            this.rateReporter = rateReporter;
             this.log = logger;
         }
 
         [HttpGet]
         public async Task<SimulationListApiModel> GetAsync()
         {
-            return new SimulationListApiModel(await this.simulationsService.GetListAsync());
+            var simulationList = await this.simulationsService.GetListAsync();
+            var simulationListApiModel = new SimulationListApiModel();
+            foreach (var x in simulationList)
+            {
+                var simulationApiModel = SimulationApiModel.FromServiceModel(x);
+                this.AppendHubPropertiesAndStatisticsAsync(simulationApiModel);
+                simulationListApiModel.Items.Add(simulationApiModel);
+            }
+
+            return simulationListApiModel;
         }
 
         [HttpGet("{id}")]
         public async Task<SimulationApiModel> GetAsync(string id)
         {
-            return SimulationApiModel.FromServiceModel(await this.simulationsService.GetAsync(id));
+            var simulation = await this.simulationsService.GetAsync(id);
+
+            if (simulation == null)
+            {
+                this.log.Warn("Simulation not found", () => new { id });
+                throw new BadRequestException("No data or invalid id provided.");
```

Comment:
should be `ResourceNotFoundException` resulting in a 404 Not Found.  This instead would lead to 400 Bad Request, which we use only when a client is broken, i.e. it has a bug and is making an invalid request